scripts/quality/run_qlty_zero.py

```
@@ -225,9 +221,7 @@ def _build_check_entry(
         has_findings = _smells_output_indicates_findings(output_tail)
     else:
         has_findings = False
-    if int(result.returncode) != 0 and name != "smells":
-        status = "fail"
-    elif name == "smells" and has_findings:
+    if int(result.returncode) != 0 and name != "smells" or name == "smells" and has_findings:
```

Ruff (0.15.11)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)